src/Nethermind/Nethermind.HealthChecks/FreeDiskSpaceChecker.cs

```diff
@@ -80,5 +65,44 @@ public async ValueTask DisposeAsync()
             await StopAsync(default);
             _timer.Dispose();
         }
+
+        public void EnsureEnoughFreeSpaceOnStart(ITimerFactory timerFactory)
+        {
+            float minAvailableSpaceThreshold = 2 * _healthChecksConfig.LowStorageSpaceShutdownThreshold;
+            if (_healthChecksConfig.LowStorageSpaceWarningThreshold > 0 && _healthChecksConfig.LowStorageSpaceWarningThreshold > _healthChecksConfig.LowStorageSpaceShutdownThreshold)
+                minAvailableSpaceThreshold = _healthChecksConfig.LowStorageSpaceWarningThreshold / 2.0f;
+
+            if (!IsEnoughDiskSpace(minAvailableSpaceThreshold))
+            {
+                ManualResetEventSlim mre = new(false);
```

Comment:
Overall not sure about the design, other clients if they can't start they just `panic` and exit the app. Maybe this should be configurable and non-default behaviour?

Reply:
Added configuration, default behaviour would be to shutdown node. New config can be used for docker setup.